sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ordering.scala

```diff
@@ -97,13 +97,18 @@ object InterpretedOrdering {
 object RowOrdering extends CodeGeneratorWithInterpretedFallback[Seq[SortOrder], BaseOrdering] {
 
   /**
-   * Returns true iff the data type can be ordered (i.e. can be sorted).
+   * Returns true if the data type can be ordered (i.e. can be sorted).
    */
-  def isOrderable(dataType: DataType): Boolean = dataType match {
+  def isOrderable(dataType: DataType,
```

Comment:
Should we fix https://github.com/apache/spark/pull/31967 first?

Reply:
@HyukjinKwon - Thanks for checking this PR. Yes we can wait for this PR https://github.com/apache/spark/pull/32552. The fix in this will work with group by, order by , partition by in window.